### PR TITLE
added context and api call, implemented watch later and liked videos …

### PR DESCRIPTION
**Pull request by nikhiljugale007**
…feature

## preview: https://deploy-preview-3--genuine-cocada-362283.netlify.app/videos

Feature completed:
1 filter by tabs
2 add to watchlater
3 remove from watch later
4 add and remove from liked videos

File to check:
apicalls.js
videocard.jsx
videocontext.js
videofilterfunction.js
videoreduckerfuncion.js

### CODE REVIEW

File: src/components/video-card/VideoCard.jsx [hunk 1]
```diff
@@ -7,12 +7,67 @@ import {
...
+			setShowDropDown((prev) => !prev);
+		} else {
+			console.log("ERROR");
+		}
+	};
+
+	const addVideoToLikedVideo = async () => {
+		const response = await addToLikedVideos(video);
+		if (response.success) {
+			console.log(response);
```
Comment (Naman-Saxena1) [inline src/components/video-card/VideoCard.jsx:47]: When you have some extra time, can replace console log statements with Toast component. For now its fine.

File: src/components/video-card/VideoCard.css [hunk 2]
```diff
@@ -1,6 +1,15 @@
...
+	-webkit-box-orient: vertical;
+	overflow: hidden;
+}
 .card {
 	position: relative;
-	min-width: 30rem;
+}
+.card-img-container {
+	height: 300px;
+	object-fit: fill;
```
Comment (Naman-Saxena1) [inline src/components/video-card/VideoCard.css:12]: object-fit: contain or scale-down is looking better.
fill is giving feeling that images are compressed.

**Review by Naman-Saxena1 — CHANGES_REQUESTED** [review]
Code looks fine 👍 
Suggested 1 change
Can also update app name from "React App" to "VercelTube" or anything of your choice and also app favicon icon.

File: src/components/video-card/VideoCard.jsx (hunk 1 above)
Comment (nikhiljugale007) [reply src/components/video-card/VideoCard.jsx:47]: I will try to change it to toast before the deadline.

File: src/api/apicalls.js [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+		const response = await axios.post("/api/auth/login", { user });
+		return { token: response.data.encodedToken, success: true };
+	} catch (err) {
+		return { token: "", success: false };
+	}
+};
+
+const getAllVideos = async () => {
+	try {
+		const response = await axios.get("/api/videos");
```
Comment (rajaranjanmr) [inline src/api/apicalls.js:25]: suggestion : here api call could have been generalised in order to maintain DRY.

**Review by rajaranjanmr — APPROVED** [review]
rest seems fine and can be mergged

File: src/components/video-card/VideoCard.css (hunk 2 above)
Comment (nikhiljugale007) [reply src/components/video-card/VideoCard.css:12]: object-fit: contain making white background around the images.so I put fill.

File: src/api/apicalls.js (hunk 3 above)
Comment (nikhiljugale007) [reply src/api/apicalls.js:25]: I will try to generalize api calls once i complete all p0. Thanks for suggestion.

**Pull request merged**